Sources/Datadog/RUMMonitor.swift

```diff
@@ -80,6 +89,41 @@ public class RUMMonitor: RUMMonitorInternal {
         stop(view: viewController, attributes: attributes)
     }
 
+    /// Notifies that an Error occurred in currently presented View.
+    /// - Parameters:
+    ///   - message: a message explaining the Error.
+    ///   - source: the origin of the Error.
+    ///   - attributes: custom attributes to attach to the Error
+    ///   - file: the file in which the Error occurred (the default is the file name in which this method was called).
+    ///   - line: the line number on which the Error occurred (the default is the line number on which this method was called).
+    public func addViewError(
+        message: String,
+        source: RUMErrorSource,
+        attributes: [AttributeKey: AttributeValue]? = nil,
+        file: StaticString = #file,
+        line: UInt = #line
```

Comment:
i think `#file` is 💯 here and i'm not sure about that but i guess our users isolate 3rd parties like us and they will call this method from a single place, to give an idea:
```swift
// MyAppRUM.swift
import Datadog
class MyRUM {
  private let _rum = Datadog.RUMMonitor(...)
  func addError(...) { _rum.addViewError(...) }
}
```
could they have wrong file/line info in that case because of isolation?

•would they get `MyAppRUM.swift` for every error?
• or we rely on the fact that practically everyone will use auto-instrumentation for RUM?

Reply:
It will be `MyAppRUM.swift: 5` in that case, but what people can do is simply:
```swift
// MyAppRUM
func addError(..., file: StaticString = #file, line: UInt = #line) {
   rumMonitor.addViewError(..., file: file, line: line)
}
```
so they will get the right source reference for free. It's like working with `XCTAsserts*` called in external matchers.

I found no better way to model it, but I wanted to have this included as it gives huge value for debugging.